src/main/java/uk/gov/hmcts/cmc/claimstore/services/ScheduledStateTransitionService.java

```diff
@@ -0,0 +1,143 @@
+        Environment environment
+    ) {
+        this.workingDayIndicator = workingDayIndicator;
+        this.caseSearchApi = caseSearchApi;
+        this.userService = userService;
+        this.caseRepository = caseRepository;
+        this.appInsights = appInsights;
+        this.emailContentProvider = emailContentProvider;
+        this.emailService = emailService;
+        this.emailProperties = emailProperties;
+        this.environment = environment;
+    }
+
+    public void stateChangeTriggered(StateTransition stateTransition) {
+        LocalDateTime now = LocalDateTime.now();
+        if (workingDayIndicator.isWorkingDay(now.toLocalDate())) {
+            User anonymousCaseWorker = userService.authenticateAnonymousCaseWorker();
+            transitionClaims(now, anonymousCaseWorker, stateTransition);
+        }
+    }
+
+    public void transitionClaims(LocalDateTime runDateTime, User user, StateTransition stateTransition) {
+        StateTransitionCalculator stateTransitionCalculator = getStateTransitionCalculator(stateTransition);
+        LocalDate responseDate = stateTransitionCalculator.calculateDateFromDeadline(runDateTime);
+
+        Set<Claim> claims = new HashSet<>(caseSearchApi.getClaims(user,
+            stateTransition.getQuery().apply(responseDate)));
+
+        Collection<Claim> failedClaims = claims.stream()
+            .map(claim -> updateClaim(user, claim, stateTransition))
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .collect(Collectors.toList());
+
+        if (!failedClaims.isEmpty()) {
+            sendFailedNotification(failedClaims, stateTransition.getCaseEvent());
+        }
+    }
+
+    private Optional<Claim> updateClaim(User user, Claim claim, StateTransition stateTransition) {
+        try {
+            appInsights.trackEvent(stateTransition.getAppInsightsEvent(), REFERENCE_NUMBER, claim.getReferenceNumber());
+            caseRepository.saveCaseEvent(user.getAuthorisation(), claim, stateTransition.getCaseEvent());
+
+            return Optional.empty();
+        } catch (Exception e) {
+            logger.error(String.format("Error whilst transitioning claim %s vis caseEvent %s",
+                claim.getReferenceNumber(), stateTransition.getCaseEvent().getValue()), e);
+            return Optional.of(claim);
+        }
+    }
+
+    private void sendFailedNotification(Collection<Claim> failedClaims, CaseEvent caseEvent) {
+        EmailContent emailContent = emailContentProvider.createContent(failedClaims, caseEvent);
+
+        EmailData emailData = new EmailData(
+            emailProperties.getRecipient(),
+            emailContent.getSubject(),
+            emailContent.getBody(),
+            Collections.emptyList()
+        );
+
+        emailService.sendEmail(emailProperties.getSender(), emailData);
+    }
+
+    private StateTransitionCalculator getStateTransitionCalculator(StateTransition stateTransition) {
```

Comment:
Somebody can remove the config for deadlineinDayas without knowing it will fail job. Would prefer them to be annotated and  fetched in this method based on stateTransition name (edited) 

Let me know your views

Reply:
Have added checks for properties in an @PostConstruct method